pkg/apis/rollouts/v1alpha1/types.go

```diff
@@ -295,6 +297,14 @@ type CanaryStrategy struct {
 	AbortScaleDownDelaySeconds *int32 `json:"abortScaleDownDelaySeconds,omitempty" protobuf:"varint,13,opt,name=abortScaleDownDelaySeconds"`
 }
 
+// Analysis configuration for the analysis runs to retain
+type Analysis struct {
+	// SucceedRunHistoryLimit limits the number of old analysis runs succeeded to be retained in a history
+	SucceedRunHistoryLimit *int32 `json:"succeedRunHistoryLimit,omitempty" protobuf:"varint,1,opt,name=succeedRunHistoryLimit"`
```

Comment:
Can we rename this to: `successfulRunHistoryLimit`  in order to be consistent with the Rollout phase `Successful`

```golang
// AnalysisPhase is the overall phase of an AnalysisRun, MetricResult, or Measurement
type AnalysisPhase string

// Possible AnalysisPhase values
const (
	AnalysisPhasePending      AnalysisPhase = "Pending"
	AnalysisPhaseRunning      AnalysisPhase = "Running"
	AnalysisPhaseSuccessful   AnalysisPhase = "Successful"
	AnalysisPhaseFailed       AnalysisPhase = "Failed"
	AnalysisPhaseError        AnalysisPhase = "Error"
	AnalysisPhaseInconclusive AnalysisPhase = "Inconclusive"
)

```

Reply:
Renamed *succeed* to *successful* everywhere for this feature